transmogrifier/sources/dspace_dim.py

```diff
@@ -0,0 +1,243 @@
+
+    def __iter__(self) -> Iterator[TimdexRecord]:
+        return self
+
+    def __next__(self) -> TimdexRecord:
+        xml = next(self.input_records)
+        record = self.create_from_dspace_dim(
+            self.source, self.source_base_url, self.source_name, xml
+        )
+        return record
+
+    @classmethod
+    def create_from_dspace_dim(
+        cls, source: str, source_base_url: str, source_name: str, xml: Tag
+    ) -> TimdexRecord:
+        """
+        Args:
+            source: A label for the source repository that is prepended to the
+            timdex_record_id.
+            source_base_url: The base URL for the source system from which direct links
+            to source metadata records can be constructed.
+            source_name: The full human-readable name of the source repository to be
+            used in the TIMDEX record.
+            xml: A BeautifulSoup Tag representing a single DSpace record in
+            dim XML.
+        """
+        # Required fields in TIMDEX
+        source_record_id = xml.header.find("identifier").string
+        all_titles = xml.find_all("dim:field", element="title")
+        main_title = [t for t in all_titles if "qualifier" not in t.attrs]
+        if len(main_title) != 1:
+            raise ValueError(
+                "A record must have exactly one title. Titles found for record "
+                f"{source_record_id}: {main_title}"
+            )
+        kwargs = {
+            "source": source_name,
+            "source_link": source_base_url + source_record_id,
+            "timdex_record_id": f"{source}:{source_record_id.replace('/', '-')}",
+            "title": main_title[0].string,
+        }
+
+        # Optional fields in TIMDEX
+        # alternate_titles, uses full title list retrieved for main title field
+        for alternate_title in [
+            t for t in all_titles if "qualifier" in t.attrs and t.string
+        ]:
+            a = AlternateTitle(
+                value=alternate_title.string,
+                kind=alternate_title.get("qualifier"),
+            )
+            kwargs.setdefault("alternate_titles", []).append(a)
+
+        # contributors
+        citation_creators = []
+        for creator in [
+            c for c in xml.find_all("dim:field", element="creator") if c.string
+        ]:
+            citation_creators.append(creator.string)
+            c = Contributor(
+                value=creator.string,
+                kind="Creator",
+            )
+            kwargs.setdefault("contributors", []).append(c)
+
+        for contributor in [
+            c for c in xml.find_all("dim:field", element="contributor") if c.string
+        ]:
+            if contributor.get("qualifier") == "author":
+                citation_creators.append(contributor.string)
+            c = Contributor(value=contributor.string, kind=contributor.get("qualifier"))
+            kwargs.setdefault("contributors", []).append(c)
+
```

Comment:
I've added this but I haven't found an usage of it in WHOAS

Reply:
Sure, but this isn't a WHOAS transform it's a dspace dim transform that we may use on other sources. For additional context: I'm thinking we might want to run the DSpace@MIT records through this transform too and see if we get better data than the mets. It turns out a LOT of dc fields aren't mapped to the OAI-PMH mets even though they are mapped in the general mets crosswalk (go figure). But inherently all dc metadata fields are mapped to dim because it's the internal format used for dspace metadata. The catch is that some data (like file formats) will only show up in dim if it's explicitly added to the dc fields, whereas it will always show up in the mets filesec. So which metadata type we want to use may ultimately depend on which fields we think are most important for a given source. Anyway that was a lot of info and I'm adding a spike ticket to investigate this question :)